js/server/modules/@arangodb/arango-collection.js

```diff
@@ -100,16 +100,24 @@ ArangoCollection.prototype.toArray = function () {
 // / @brief was docuBlock collectionTruncate
 // //////////////////////////////////////////////////////////////////////////////
 
-ArangoCollection.prototype.truncate = function () {
+ArangoCollection.prototype.truncate = function (options) {
+  if (typeof options === 'boolean') {
+    options = { waitForSync: options || false };
+  } else {
+    options = options || {};
+  }
+  if (!options.hasOwnProperty('compact')) {
+    options.compact = true;
+  }
   var cluster = require('@arangodb/cluster');
   if (cluster.isCoordinator()) {
     if (this.status() === ArangoCollection.STATUS_UNLOADED) {
       this.load();
     }
   }
-  var ret = this.TRUNCATE();
+  var ret = this.TRUNCATE(options.compact);
```

Comment:
You add handling for `waitForSync` here, like in the `js/client` code - seems good to make this consistent. However, `waitForSync` is never passed to `TRUNCATE` and just ignored.